app/main/views/drafts.py

```diff
@@ -222,15 +223,17 @@ def publish_draft_service(draft_id):
         DraftService.id == draft_id
     ).first_or_404()
 
+    if draft.status == 'not-submitted':
+        abort(400, "Cannot publish a draft if it is not submitted: {}".format(draft.status))
+    if draft.status == 'submitted' and draft.service_id:
+        abort(400, "Cannot re-publish a submitted service")
+
     if draft.service_id:
         service = Service.query.filter(
             Service.service_id == draft.service_id
         ).first_or_404()
 
-        service_from_draft = update_and_validate_service(
-            service,
-            draft.data)
-
+        service_from_draft = update_and_validate_service(service, draft.data)
     else:
         service_from_draft = create_service_from_draft(draft, "enabled")
```

Comment:
While this helps with not indexing the G7 drafts before the framework is live, that's not quite what we want. New G7 services should be "published" since we won't have an additional step to update all the service statuses, and we need an additional bit of logic that prevents them from being sent to search API when promoted from submitted drafts